src/utils/BeameUtils.js

```diff
@@ -88,6 +87,77 @@ function findInTree(node, predicate, limit) {
 }
 //endregion
 
+//region Background Jobs
+/**
+ * @typedef {Object} BackgroundJob
+ * @property {Object} handle => set interval handle
+ * @property {Number} interval => interval to run the job
+ * @property {Number} called => times the job was called
+ * @property {Number} failed => times the job failed
+ * @property {Boolean} running => will be true while the process is running
+ * @property { {result: * | undefined, error: * | undefined, timestamp: Number} | undefined} lastRun => result from the last run
+ */
+/**
+ * @type {{name: BackgroundJob}}
+ */
+const backgroundJobs = {};
+/**
+ * Starts a background job that will run every interval
+ * @param {String} name
+ * @param {Function} func
+ * @param {Number} interval (in ms)
+ * @returns {boolean}
+ */
+function startBackgroundJob(name, func, interval) {
+	if(backgroundJobs[name]) {
+		logger.warn(`Background job '${name}' is already running, skipping....`);
+		return false;
+	}
+
+	async function jobCall() {
+		backgroundJobs[name].called++;
+		backgroundJobs[name].running = true;
+		try {
+			backgroundJobs[name].lastRun = { result: await func(), timestamp: Date.now() };
+		}
+		catch(e) {
```

Comment:
I though you added printing of the exception

Reply:
ups ;) 👍